src/librustc/hir/mod.rs

```diff
@@ -1450,7 +1450,7 @@ pub struct ItemId {
 /// The name might be a dummy name in case of anonymous items
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub struct Item {
-    pub name: Name,
+    pub name: Spanned<Name>,
```

Comment:
While waiting for @nikomatsakis 

@KiChjang if we add a `span` to `ast::Ident` then we likely to change the signature of this method [here](https://github.com/rust-lang/rust/blob/1265cbf4e05628c98f51afebe0b662c451173faa/src/libsyntax/ast.rs#L77-L79).

```rust
pub const fn with_empty_ctxt(name: Name) -> Ident {
    Ident { name: name, ctxt: SyntaxContext::empty() }
}
```

And this will require to change in many places [where-ever](https://github.com/rust-lang/rust/search?utf8=%E2%9C%93&q=%22Ident%3A%3Awith_empty_ctxt%22&type=Code) the method is called.

Should it be ok?